packages/generator/ni_measurement_plugin_sdk_generator/client/__init__.py

```diff
@@ -60,89 +61,110 @@ def _create_file(
+                    print(
+                        f"Warning: The service class '{service_class}' does not follow the recommended format."
+                    )
+
+        if not is_python_identifier(module_name):
             raise click.ClickException(
-                "Unable to create client.\nPlease provide a valid module name or update the measurement with a valid service class."
+                f"The module name '{module_name}' is not a valid Python identifier."
+            )
+        if not is_python_identifier(class_name):
+            raise click.ClickException(
+                f"The class name '{class_name}' is not a valid Python identifier."
             )
 
-        if module_name is None:
-            module_name = camel_to_snake_case(base_service_class) + "_client"
-        if class_name is None:
-            class_name = base_service_class.replace("_", "") + "Client"
-            if not any(ch.isupper() for ch in base_service_class):
-                print(
-                    f"Warning: The service class '{measurement_service_class}' does not follow the recommended format."
-                )
-
-    if not module_name.isidentifier():
-        raise click.ClickException(
-            f"The module name '{module_name}' is not a valid Python identifier."
+        measurement_service_stub = get_measurement_service_stub(
+            discovery_client, channel_pool, service_class
         )
-    if not is_python_identifier(class_name):
-        raise click.ClickException(
-            f"The class name '{class_name}' is not a valid Python identifier."
+        metadata = measurement_service_stub.GetMetadata(
+            v2_measurement_service_pb2.GetMetadataRequest()
         )
+        configuration_metadata = get_configuration_metadata_by_index(metadata, service_class)
+        output_metadata = get_output_metadata_by_index(metadata)
 
-    measurement_service_stub = get_measurement_service_stub(
-        discovery_client, channel_pool, measurement_service_class
-    )
-    metadata = measurement_service_stub.GetMetadata(v2_measurement_service_pb2.GetMetadataRequest())
-    configuration_metadata = get_configuration_metadata_by_index(
-        metadata, measurement_service_class
-    )
-    output_metadata = get_output_metadata_by_index(metadata)
+        configuration_parameters_with_type_and_default_values, measure_api_parameters = (
+            get_configuration_parameters_with_type_and_default_values(
+                configuration_metadata, built_in_import_modules
+            )
+        )
+        output_parameters_with_type = get_output_parameters_with_type(
+            output_metadata, built_in_import_modules, custom_import_modules
+        )
 
-    configuration_parameters_with_type_and_default_values, measure_api_parameters = (
-        get_configuration_parameters_with_type_and_default_values(
-            configuration_metadata, built_in_import_modules
+        if directory_out is None:
```

Comment:
Can we move this directory validation outside the for loop? As it is common for all the services